include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -7018,6 +7150,49 @@ FEEvaluation<dim,
 
 
 
+template <int dim,
+          int fe_degree,
+          int n_q_points_1d,
+          int n_components_,
+          typename Number,
+          typename VectorizedArrayType>
+inline void
+FEEvaluation<
+  dim,
+  fe_degree,
+  n_q_points_1d,
+  n_components_,
+  Number,
+  VectorizedArrayType>::evaluate(const VectorizedArrayType * values_array,
+                                 const EvaluationFlags::type evaluation_flags)
+{
+  SelectEvaluator<dim,
+                  fe_degree,
+                  n_q_points_1d,
+                  n_components,
+                  VectorizedArrayType>::
+    evaluate(*this->data,
+             const_cast<VectorizedArrayType *>(values_array),
+             this->values_quad[0],
+             this->gradients_quad[0][0],
+             this->hessians_quad[0][0],
+             this->scratch_data,
+             evaluation_flags & EvaluationFlags::values,
+             evaluation_flags & EvaluationFlags::gradients,
+             evaluation_flags & EvaluationFlags::hessians);
+
+#  ifdef DEBUG
+  if (evaluation_flags & EvaluationFlags::values)
+    this->values_quad_initialized = true;
```

Comment:
You might as well replace these three member variables with your enum as well :-)

Reply:
I can take those internal parts over.